invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/Identifiers/PIDField/components/OptionalDOIoptions.js

```diff
@@ -23,24 +23,36 @@ export class OptionalDOIoptions extends Component {
   };
 
   _render = (cmp, shouldWrapPopup, message) =>
-    shouldWrapPopup ? <Popup content={message} trigger={cmp} /> : cmp;
+    shouldWrapPopup && message ? <Popup content={message} trigger={cmp} /> : cmp;
 
   render() {
-    const { isManagedSelected, isNoNeedSelected, pidLabel, optionalDOItransitions } =
-      this.props;
-
-    const allDOIoptionsAllowed = _isEmpty(optionalDOItransitions);
-    const isUnManagedDisabled =
-      isManagedSelected ||
-      (!allDOIoptionsAllowed &&
-        !optionalDOItransitions.allowed_providers.includes("external"));
-    const isNoNeedDisabled =
-      isManagedSelected ||
-      (!allDOIoptionsAllowed &&
-        !optionalDOItransitions.allowed_providers.includes("not_needed"));
+    const {
+      isManagedSelected,
+      isNoNeedSelected,
+      pidLabel,
+      optionalDOItransitions,
+      record,
+    } = this.props;
+
+    const doi = record?.pids?.doi?.identifier || "";
+    const isNewUpload = record.status === "draft";
```

Comment:
Minor: We could user [this](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/state/reducers/deposit.js#L35) to avoid hardcoding the state.

Also is `isNewUpload` a correct name?  just find it a bit confusing 😅 Does this mean that after the first version is published if we create a new version it doesn't has the draft status?

Reply:
no actually it has `status=='new_version_draft'`. Thanks for the suggestion, I will integrate it now!